apis/webhook/v1alpha1/types.go

```diff
@@ -0,0 +1,137 @@
+/*
+Copyright 2021 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	xpv1 "github.com/crossplane/crossplane-runtime/apis/common/v1"
+	"github.com/crossplane/provider-bitbucket-server/internal/clients/bitbucket"
+)
+
+/*
+https://docs.atlassian.com/bitbucket-server/rest/7.10.0/bitbucket-rest.html
+
+    "active": true
+    }
+*/
+
+// WebhookParameters are the configurable fields of a Webhook.
+type WebhookParameters struct {
+	// The project key is the short name for the project for a
+	// repository. Typically the key for a project called "Foo Bar"
+	// would be "FB".
+	// +immutable
+	ProjectKey string `json:"projectKey"`
+
+	// The repoName is the name of the git repository.
+	// +immutable
+	RepoName string `json:"repoName"`
+
+	Webhook BitbucketWebhook `json:"webhook"`
+}
+
+type BitbucketWebhook struct {
+	// +immutable
+	Name string `json:"name"`
+
+	// +immutable
```

Comment:
fiels here and in secret are not immutable, this service has a proper update API